### PR TITLE
Add ambient sensing demo

### PR DESCRIPTION
**Pull request by farshidtz**
- [x] Install platform
- [x] Sensing
- [x] Setup device mqtt
- [x] Query data
- [x] Setup grafana dash
- [x] Create model assertion and gadget

### CODE REVIEW

File: Ambient-Sensing.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+- DHT22/DHT11 temperature and humidity sensor modules
+
+We use the following:
+- Raspberry Pi 4 running arm64 Ubuntu Core (hostname: `jupiter.local`)
+- Raspberry Pi Zero W running armhf Raspberry Pi OS (hostname: `pluto.local`)
+- DHT22 sensor connected to the Raspberry Pi Zero W
+
+To interact with the APIs, we use the [HTTPie HTTP Client](https://snapcraft.io/httpie) which provides the `http` command. It is easy to use, prints pretty JSON output, and is available as a Snap! You can use any other HTTP client.
+
+## Installing the EdgeX platform
```
Comment (farshidtz) [inline Ambient-Sensing.md:17]: Need to make it clear which step is related to which device.

File: Ambient-Sensing.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+sudo snap restart edgexfoundry.security-secretstore-setup
+```
+
+## Setting up sensing input
+
+We assume that sensors have been setup and the measurements for temperature and humidity are published to an MQTT broker at an interval. Setting up sensors and publishing their data to the broker is beyond the scope of this demo. 
+
+We use the following:
+- [Mosquitto](https://snapcraft.io/mosquitto) MQTT broker on `jupiter.local`
+- [DHT-MQTT](https://github.com/farshidtz/dht-mqtt) for reading data DHT11/DHT22 sensors on `pluto.local` and publishing to the broker. The configuration is as follows:
```
Comment (MonicaisHer) [inline Ambient-Sensing.md:42]: https://github.com/farshidtz/dht-mqtt/blob/a99ecd9cef1ba0c85b77daae76b5976a8ecc7628/run.sh#L10

Here is an error when I trying to run DHT-MQTT with Raspberry Pi 4B:
```
sudo ./run.sh 
Traceback (most recent call last):
  File "/home/pi/Desktop/dht-mqtt/dht-mqtt.py", line 10, in <module>
    import adafruit_dht
ModuleNotFoundError: No module named 'adafruit_dht'
```
Maybe `python3 -u $DIR/dht-mqtt.py` would solve this error?

Comment (farshidtz) [reply Ambient-Sensing.md:42]: Latest OS revisions come with Python 3 by default. Perhaps you are using an older one?

Try installing the python-is-python3 package.

Comment (MonicaisHer) [reply Ambient-Sensing.md:42]: Thanks, you ate right, I used an old Raspberry pi OS. I have updated it and DHT-MQTT works now.
I also need to install paho:
```
pip3 install paho-mqtt
```

File: Ambient-Sensing.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+```
+Those files are still available as read-only under `/snap/edgex-device-mqtt/current/config/device-mqtt/res/`.
+
+Add `profiles/temperature-humidity-sensor.yml`:
+```yml
+name: "temperature-humidity-sensor"
+deviceResources:
+- name: temperature
+  properties:
+    valueType: "Float32"
```
Comment (MonicaisHer) [inline Ambient-Sensing.md:94]: Why not use `valueType: "int32"` in this case? Or there should be a calculation transform from float32 to int in Grafana.

File: Ambient-Sensing.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+>
+> ```bash
+> # Delete device:
+> http DELETE http://localhost:59881/api/v2/device/name/pluto-dht22
+>
+> # Delete profile, if modified:
+> http DELETE http://localhost:59881/api/v2/deviceprofile/name/temperature-humidity-sensor
+>
+> # Restart:
+> sudo snap restart edgex-device-usb-camera
```
Comment (MonicaisHer) [inline Ambient-Sensing.md]: sudo snap restart edgex-device-mqtt

File: Ambient-Sensing.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+> ℹ **Tip**  
+> The range can be shorted by manually entering the from value such as: now-1m
+
+#### Setup the panel
+a. Add an empty panel. Set the title to Temperature.
+
+b. Setup query and transformation:
+-   Name `Pluto`
+-   Field `$.readings[:].value`, Type `Boolean`, Alias `Value`
```
Comment (MonicaisHer) [inline Ambient-Sensing.md:230]: The value's type should be either string(float32) or number(int32)

File: Ambient-Sensing.md (hunk 3 above)
Comment (farshidtz) [reply Ambient-Sensing.md:94]: Since the sensor readings have fractions, we should use float here.

File: Ambient-Sensing.md (hunk 2 above)
Comment (farshidtz) [reply Ambient-Sensing.md:42]: Thanks. Changed to python3 (to make it compatible with older OSes) and added paho.

https://github.com/farshidtz/dht-mqtt/commit/83b6d1c319382e192afe56e3346543e55a025aad

File: Ambient-Sensing.md (hunk 1 above)
Comment (farshidtz) [reply Ambient-Sensing.md:17]: Fixed

File: Ambient-Sensing.md (hunk 5 above)
Comment (farshidtz) [reply Ambient-Sensing.md:230]: Removed the grafana section for now.

**Comment by farshidtz** [comment]
Merged into https://github.com/canonical/edgex-demos/wiki

**Pull request closed without merging**